org-code-javabuilder/theater/src/main/java/org/code/theater/GifWriter.java

```diff
@@ -48,6 +50,13 @@ public GifWriter createGifWriter(ByteArrayOutputStream out) {
    */
   public void writeToGif(BufferedImage img, int delay) {
     try {
+      if (this.imageOutputStream.length() > MAX_STREAM_LENGTH_BYTES) {
+        Theater.stage.close();
+        // TODO: Make this translatable: https://codedotorg.atlassian.net/browse/CSA-1108
+        String message =
+            "Your video is too large. Please decrease the number of frames in your video and try again.";
+        throw new RuntimeException(message);
```

Comment:
InternalException or UserInitiatedException?

Reply:
I think I'll keep this as a runtime exception until we add the translation piece on the frontend, if we do an InternalException I think we'll lose the message.